pkg/storage/ingest/pusher.go

```diff
@@ -75,14 +79,32 @@ func (c pusherConsumer) pushRequests(ctx context.Context, reqC <-chan parsedReco
 
 		c.processingTimeSeconds.Observe(time.Since(processingStart).Seconds())
 		if err != nil {
-			level.Error(c.l).Log("msg", "failed to push write request; skipping", "err", err)
-			// TODO move distributor's isClientError to a separate package and use that here to swallow only client errors and abort on others
-			continue
+			if !isClientIngesterError(err) {
+				return fmt.Errorf("consuming record at index %d for tenant %s: %w", recordIdx, wr.tenantID, err)
```

Comment:
What's the actual value of knowing the index? Looks something unpredictable so I can't understand what we'll use it for.

Reply:
my idea was to give enough information to be able to deduce the exact kafka record offset from the errors here and the error in the retry 

https://github.com/grafana/mimir/blob/a5d37b9717fe4ad478333ae701eded7648fcd92c/pkg/storage/ingest/reader.go#L185-L189